server/apiserver/argoserver.go

```diff
@@ -158,7 +162,10 @@ func (as *argoServer) Run(ctx context.Context, port int) {
 	go func() { as.checkServeErr("grpcServer", grpcServer.Serve(grpcL)) }()
 	go func() { as.checkServeErr("httpServer", httpServer.Serve(httpL)) }()
 	go func() { as.checkServeErr("tcpm", tcpm.Serve()) }()
-	log.Infof("Argo Server started successfully on address %s", address)
+	log.Infof("Argo Server started successfully on address http://locahost%s", address)
+	if !as.disableOpenBrowser {
```

Comment:
this is the wrong level of abstraction, I don't think you should be opening the browser so deep in the code, instead do it in cmd/argo/server  - you'll need to use a callback to achieve this and this func is blocking